2018-01-22-javascript-hard-parts/fem-JavaScriptTheHardParts-ClosuresSolutions.js

```diff
@@ -104,8 +104,10 @@ afterCalled(); // -> 'hello' is printed
 // EXTENSION: CHALLENGE 7 //
 
 function delay(func, wait) {
-    setTimeout(func, wait);
-}
+    var args = Array.from(arguments);
+    if (args.length > 2) { args = args.slice(2, args.length) }
+    setTimeout(function() { func.apply(null,args) }, wait);
+} 
```

Comment:
A possibly cleaner way to write this with destructuring would be:

```js
function delay() {
  let [func, wait, ...args] = Array.from(arguments);
  setTimeout(function() { func.apply(null, args) }, wait);
}
```

Reply:
The down side of using the destructuring is that its not as clear that `func` and `wait` are required args.

Reply:
Oh! You can do this (with the leading commas in the destructuring):

```js
function delay(func, wait) {
  let [, , ...args] = Array.from(arguments);
  setTimeout(function() { func.apply(null, args) }, wait);
}
```